### ISSUE TITLE
OpenBSD/LibreSSL build issues

### ISSUE THREAD

**Issue by protoCall7** [sfackler/rust-openssl#1167]
After issue #1121 was solved, openssl-sys 0.9.49 was building fine with LibreSSL 3.0 however I noticed during a build today that both 0.9.49 and 0.9.50 are refusing to support LibreSSL > 2.9.

Is this a regression, or am I possibly doing something wrong in my build?

```
$ sysctl kern.version
kern.version=OpenBSD 6.6 (GENERIC.MP) #344: Wed Oct  2 11:48:47 MDT 2019
    deraadt@amd64.openbsd.org:/usr/src/sys/arch/amd64/compile/GENERIC.MP

$ openssl version
LibreSSL 3.0.1

     Running `/usr/obj/ports/zola-0.9.0/build-amd64/target/release/build/openssl-sys-8b37e69ffe54066f/build-script-main`
error: failed to run custom build command for `openssl-sys v0.9.50`

Caused by:
  process didn't exit successfully: `/usr/obj/ports/zola-0.9.0/build-amd64/target/release/build/openssl-sys-8b37e69ffe54066f/build-script-main` (exit code: 101)
--- stdout
cargo:rustc-cfg=const_fn
cargo:rerun-if-env-changed=X86_64_UNKNOWN_OPENBSD_OPENSSL_LIB_DIR
X86_64_UNKNOWN_OPENBSD_OPENSSL_LIB_DIR unset
cargo:rerun-if-env-changed=OPENSSL_LIB_DIR
OPENSSL_LIB_DIR unset
cargo:rerun-if-env-changed=X86_64_UNKNOWN_OPENBSD_OPENSSL_INCLUDE_DIR
X86_64_UNKNOWN_OPENBSD_OPENSSL_INCLUDE_DIR unset
cargo:rerun-if-env-changed=OPENSSL_INCLUDE_DIR
OPENSSL_INCLUDE_DIR unset
cargo:rerun-if-env-changed=X86_64_UNKNOWN_OPENBSD_OPENSSL_DIR
X86_64_UNKNOWN_OPENBSD_OPENSSL_DIR unset
cargo:rerun-if-env-changed=OPENSSL_DIR
OPENSSL_DIR unset
cargo:rustc-link-lib=ssl
cargo:rustc-link-lib=crypto
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-unknown-openbsd")
HOST = Some("x86_64-unknown-openbsd")
CC_x86_64-unknown-openbsd = None
CC_x86_64_unknown_openbsd = None
HOST_CC = None
CC = None
CFLAGS_x86_64-unknown-openbsd = None
CFLAGS_x86_64_unknown_openbsd = None
HOST_CFLAGS = None
CFLAGS = Some("-O2 -pipe")
CRATE_CC_NO_DEFAULTS = None
DEBUG = Some("false")
CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-O2" "-pipe" "-E" "build/expando.c"
exit code: 0
cargo:rustc-cfg=osslconf="OPENSSL_NO_BUF_FREELISTS"
cargo:rustc-cfg=osslconf="OPENSSL_NO_COMP"
cargo:rustc-cfg=osslconf="OPENSSL_NO_KRB5"
cargo:rustc-cfg=osslconf="OPENSSL_NO_PSK"
cargo:rustc-cfg=osslconf="OPENSSL_NO_RFC3779"
cargo:rustc-cfg=osslconf="OPENSSL_NO_SRP"
cargo:rustc-cfg=osslconf="OPENSSL_NO_SSL3_METHOD"
cargo:conf=OPENSSL_NO_BUF_FREELISTS,OPENSSL_NO_COMP,OPENSSL_NO_KRB5,OPENSSL_NO_PSK,OPENSSL_NO_RFC3779,OPENSSL_NO_SRP,OPENSSL_NO_SSL3_METHOD
cargo:rustc-cfg=libressl
cargo:rustc-cfg=libressl251
cargo:rustc-cfg=libressl261
cargo:rustc-cfg=libressl270
cargo:rustc-cfg=libressl271
cargo:rustc-cfg=libressl273
cargo:rustc-cfg=libressl280
cargo:rustc-cfg=libressl281
cargo:rustc-cfg=libressl291
cargo:libressl_version_number=3000100f

--- stderr
thread 'main' panicked at '

This crate is only compatible with OpenSSL 1.0.1 through 1.1.1, or LibreSSL 2.5
through 2.9.x, but a different version of OpenSSL was found. The build is now aborting
due to this version mismatch.

', /usr/obj/ports/zola-0.9.0/zola-0.9.0/modcargo-crates/openssl-sys-0.9.50/build/main.rs:241:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by sfackler** [comment]
You need https://github.com/sfackler/rust-openssl/pull/1166. I'll cut a release with it tonight.

**Comment by protoCall7** [comment]
Sounds good, thanks for the quick reply @sfackler

**Comment by sfackler** [comment]
Published in 0.9.51.

**Comment by protoCall7** [comment]
Confirmed working now. Thank you!
